std/utf.d

```diff
@@ -3490,19 +3490,26 @@ int impureVariable;
 }
 
 /****************************
- * Iterate an input range of characters by char type C.
+ * Iterate an input range of characters by char type `C` by
+ * encoding the elements of the range.
  *
- * UTF sequences that cannot be converted to UTF-8 are replaced by U+FFFD
- * per "5.22 Best Practice for U+FFFD Substitution" of the Unicode Standard 6.2.
- * Hence byUTF is not symmetric.
+ * UTF sequences that cannot be converted to the specified encoding are
+ * replaced by U+FFFD per "5.22 Best Practice for U+FFFD Substitution"
+ * of the Unicode Standard 6.2. Hence byUTF is not symmetric.
  * This algorithm is lazy, and does not allocate memory.
- * Purity, nothrow, and safety are inferred from the r parameter.
+ * `@nogc`, `pure`-ity, `nothrow`, and `@safe`-ty are inferred from the
+ * `r` parameter.
  *
  * Params:
- *      C = char, wchar, or dchar
+ *      C = `char`, `wchar`, or `dchar`
  *      r = input range of characters, or array of characters
  * Returns:
- *      input range of type C
+ *      If r is a range and not auto-decodable, as defined by
+ *      $(REF isAutodecodableString, std, traits), then a forward range is
+ *      returned if the base range is also a forward range, if not this
+ *      returns an input range of characters. Or, if r is a range and it
+ *      is auto-decodable and `is(ElementEncodingType!typeof(r) == C)`,
+ *      then the range is passed to $(LREF byCodeUnit).
```

Comment:
What about something along the lines of:
````
Returns:
    A forward range, if ... blah blah blah;

    An input range, if ... blah blah blah;

    Otherwise, returns ... blah blah blah.
````
That way it's clear what the set of possible return types are from a glance, and then you can read more carefully about the detailed conditions that would return that particular type.